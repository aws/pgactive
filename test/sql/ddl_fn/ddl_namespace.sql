\c postgres
SELECT bdr.bdr_replicate_ddl_command($DDL$
CREATE SCHEMA test_schema_1
       CREATE UNIQUE INDEX abc_a_idx ON abc (a)

       CREATE VIEW abc_view AS
              SELECT a+1 AS a, b+1 AS b FROM abc

       CREATE TABLE abc (
              a serial,
              b int UNIQUE
       );
$DDL$);
SELECT bdr.bdr_replicate_ddl_command($DDL$
CREATE FUNCTION test_schema_1.abc_func() RETURNS void
       AS $$ BEGIN END; $$ LANGUAGE plpgsql;
$DDL$);
SELECT bdr.bdr_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
\c regression
SELECT COUNT(*) FROM pg_class WHERE relnamespace =
    (SELECT oid FROM pg_namespace WHERE nspname = 'test_schema_1');

INSERT INTO test_schema_1.abc DEFAULT VALUES;
INSERT INTO test_schema_1.abc DEFAULT VALUES;
INSERT INTO test_schema_1.abc DEFAULT VALUES;

SELECT bdr.bdr_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
\c postgres
SELECT * FROM test_schema_1.abc;
SELECT * FROM test_schema_1.abc_view;

SELECT bdr.bdr_replicate_ddl_command($DDL$ ALTER SCHEMA test_schema_1 RENAME TO test_schema_renamed; $DDL$);

SELECT bdr.bdr_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
\c regression
SELECT COUNT(*) FROM pg_class WHERE relnamespace =
    (SELECT oid FROM pg_namespace WHERE nspname = 'test_schema_1');
\set VERBOSITY terse
SELECT bdr.bdr_replicate_ddl_command($DDL$ CREATE SCHEMA test_schema_renamed; $DDL$); -- fail, already exists
SELECT bdr.bdr_replicate_ddl_command($DDL$ CREATE SCHEMA IF NOT EXISTS test_schema_renamed; $DDL$); -- ok with notice

SELECT bdr.bdr_replicate_ddl_command($DDL$ DROP SCHEMA test_schema_renamed CASCADE; $DDL$);

SELECT bdr.bdr_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
\c postgres
SELECT COUNT(*) FROM pg_class WHERE relnamespace =
    (SELECT oid FROM pg_namespace WHERE nspname = 'test_schema_renamed');
