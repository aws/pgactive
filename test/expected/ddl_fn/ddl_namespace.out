\c postgres
SELECT pgactive.pgactive_replicate_ddl_command($DDL$
CREATE SCHEMA test_schema_1
       CREATE UNIQUE INDEX abc_a_idx ON abc (a)

       CREATE VIEW abc_view AS
              SELECT a+1 AS a, b+1 AS b FROM abc

       CREATE TABLE abc (
              a serial,
              b int UNIQUE
       );
$DDL$);
 pgactive_replicate_ddl_command 
--------------------------------
 
(1 row)

SELECT pgactive.pgactive_replicate_ddl_command($DDL$
CREATE FUNCTION test_schema_1.abc_func() RETURNS void
       AS $$ BEGIN END; $$ LANGUAGE plpgsql;
$DDL$);
 pgactive_replicate_ddl_command 
--------------------------------
 
(1 row)

SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
 pgactive_wait_for_slots_confirmed_flush_lsn 
---------------------------------------------
 
(1 row)

\c regression
SELECT COUNT(*) FROM pg_class WHERE relnamespace =
    (SELECT oid FROM pg_namespace WHERE nspname = 'test_schema_1');
 count 
-------
     5
(1 row)

INSERT INTO test_schema_1.abc DEFAULT VALUES;
INSERT INTO test_schema_1.abc DEFAULT VALUES;
INSERT INTO test_schema_1.abc DEFAULT VALUES;
SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
 pgactive_wait_for_slots_confirmed_flush_lsn 
---------------------------------------------
 
(1 row)

\c postgres
SELECT * FROM test_schema_1.abc;
 a | b 
---+---
 1 |  
 2 |  
 3 |  
(3 rows)

SELECT * FROM test_schema_1.abc_view;
 a | b 
---+---
 2 |  
 3 |  
 4 |  
(3 rows)

SELECT pgactive.pgactive_replicate_ddl_command($DDL$ ALTER SCHEMA test_schema_1 RENAME TO test_schema_renamed; $DDL$);
 pgactive_replicate_ddl_command 
--------------------------------
 
(1 row)

SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
 pgactive_wait_for_slots_confirmed_flush_lsn 
---------------------------------------------
 
(1 row)

\c regression
SELECT COUNT(*) FROM pg_class WHERE relnamespace =
    (SELECT oid FROM pg_namespace WHERE nspname = 'test_schema_1');
 count 
-------
     0
(1 row)

\set VERBOSITY terse
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ CREATE SCHEMA test_schema_renamed; $DDL$); -- fail, already exists
ERROR:  schema "test_schema_renamed" already exists
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ CREATE SCHEMA IF NOT EXISTS test_schema_renamed; $DDL$); -- ok with notice
NOTICE:  schema "test_schema_renamed" already exists, skipping
 pgactive_replicate_ddl_command 
--------------------------------
 
(1 row)

SELECT pgactive.pgactive_replicate_ddl_command($DDL$ DROP SCHEMA test_schema_renamed CASCADE; $DDL$);
NOTICE:  drop cascades to 3 other objects
 pgactive_replicate_ddl_command 
--------------------------------
 
(1 row)

SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
 pgactive_wait_for_slots_confirmed_flush_lsn 
---------------------------------------------
 
(1 row)

\c postgres
SELECT COUNT(*) FROM pg_class WHERE relnamespace =
    (SELECT oid FROM pg_namespace WHERE nspname = 'test_schema_renamed');
 count 
-------
     0
(1 row)

